lib/api/include/irods/get_resource_info_for_operation.h

```diff
@@ -0,0 +1,31 @@
+#ifndef IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+#define IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+
+struct RcComm;
+struct DataObjInp;
+
+/**
+ * \brief Get preferred-resource info for the specified operation.
+ * \param[in] _conn - A rcComm_t connection handle to the server.
+ * \param[in] _dataObjInp - generic dataObj input. Relevant items are:
+ *      \n      \b objPath - the path of the target data object.
+ *      \n      \b condInput - conditional Input
+ *      \n              -- REPL_NUM_KW  - The replica number of the copy to upload.
+ *      \n              -- GET_RESOURCE_INFO_FOR_OPERATION_KW - The *_OPERATION string: "CREATE", "WRITE", "OPEN", or "UNLINK".
+ *      \n              -- RESC_NAME_KW - The default destination resource. Only used
+ *                              to create a new file, no overwrite of existing files.
+ * \param[out] _out_info - a JSON string with keys "host" and "resc_hier", identifying target server and resource hierarchy.
+ *                         Example:
+ *                         {
+ *                             "host": <string>,
+ *                             "resc_hier": <string>
+ *                         }
+ * \return integer
+ * \retval 0 on success.
+**/
```

Comment:
Prefer using `///` for doxygen documentation instead of javadoc style comments.

Please format like other new API endpoints. See the following headers for examples.
- https://github.com/irods/irods/blob/f3ccaa842218e477395ebcf553639134433b63ee/lib/api/include/irods/switch_user.h
- https://github.com/irods/irods/blob/f3ccaa842218e477395ebcf553639134433b63ee/lib/api/include/irods/dataObjChksum.h
- https://github.com/irods/irods/blob/f3ccaa842218e477395ebcf553639134433b63ee/lib/api/include/irods/data_object_finalize.h
- https://github.com/irods/irods/blob/f3ccaa842218e477395ebcf553639134433b63ee/lib/api/include/irods/check_auth_credentials.h

Reply:
I'll wait for further comments to make sure this looks right before copying over to `rs_*.hpp`

Reply:
Let me know if this looks generally good now in the` rc_*` call doxygen section.  Then I'll duplicate (with all advisable mods) over to  the `rs_*` header....

Reply:
Docs duplicated to rs_* header